src/rpc/blockchain.cpp

```diff
@@ -2441,15 +2452,20 @@ static RPCHelpMan scanblocks()
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
 {
+    static GlobalMutex cs_relevant_blocks;
+    static UniValue relevant_blocks GUARDED_BY(cs_relevant_blocks);
+
     UniValue ret(UniValue::VOBJ);
     if (request.params[0].get_str() == "status") {
         BlockFiltersScanReserver reserver;
+        LOCK(cs_relevant_blocks);
```

Comment:
In commit "rpc: add relevant_blocks to scanblocks status" (4c9dc4bbe6b23ae7dba7badbdb6dd6c3d0f74e08)

Lock order here seems a little subtle. Would be good to have a comment saying it is important to lock this before calling reserve so if a scan is currently happening but is about to finish, an accurate "relevant_blocks" value can be returned before the scanning thread clears it.

Reply:
Agreed. Comment will be added.